core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -3500,4 +3500,12 @@ public static void updateNullValueBasedOnDatatype(DataOutputStream dataOutputStr
       dataOutputStream.write(CarbonCommonConstants.EMPTY_BYTE_ARRAY);
     }
   }
+
+  /**
+   * returns whether column is complex column based on column name for child column
+   * @return true if column is complex
+   */
+  public static boolean isComplexColumn(String colName) {
+    return colName.contains(".val") || colName.contains(".");
```

Comment:
1. use constant for `.`
2. The same code is present in `ColumnsSchema.java`, so you can call this method from column schema class also and keep only one method

Reply:
ok